### ISSUE TITLE
Crash while adjusting effect slider

### ISSUE THREAD

**Issue by EQXTFL** [OpenShot/openshot-qt#3718]
**Description**
Opening Pixelate effect controls, Left Margin is set to 0, left click-hold into slide area and move mouse right results in many different crashes, examples below.

**Example 1**
properties_model:INFO looping points: co.X = 1
properties_model:INFO looping points: co.X = 35
properties_model:INFO looping points: co.X = 644
properties_model:INFO updating point: co.X = 644 to value: 0.3017241379310345
Caught signal 11 (SIGSEGV)
---- Unhandled Exception: Stack Trace ----
malloc(): mismatching next->prev_size (unsorted)
Caught signal 6 (SIGABRT)
---- Unhandled Exception: Stack Trace ----

**Example 2**
properties_model:INFO looping points: co.X = 1
properties_model:INFO looping points: co.X = 35
properties_model:INFO looping points: co.X = 635
properties_model:INFO updating point: co.X = 635 to value: 0.3793103448275862
Caught signal 11 (SIGSEGV)
---- Unhandled Exception: Stack Trace ----
munmap_chunk(): invalid pointer
Caught signal 6 (SIGABRT)
---- Unhandled Exception: Stack Trace ----
Segmentation fault (core dumped)

**Example 3**
preview_thread:INFO refreshFrame
preview_thread:INFO self.player.Position(): 635
Caught signal 11 (SIGSEGV)
---- Unhandled Exception: Stack Trace ----
  /usr/lib/x86_64-linux-gnu/libopenshot.so.16 ( openshot::Pixelate::GetFrame(std::shared_ptr<openshot::Frame>, long)  + 0x504 )  [0x7fd6c02307a0]
  /usr/lib/x86_64-linux-gnu/libopenshot.so.16 ( openshot::Clip::apply_effects(std::shared_ptr<openshot::Frame>)  + 0xd8  )  [0x7fd6c0203db8]
  /usr/lib/x86_64-linux-gnu/libopenshot.so.16 ( openshot::Clip::GetFrame(long)            + 0x5e1 )  [0x7fd6c01f8f3b]
  /usr/lib/x86_64-linux-gnu/libopenshot.so.16 ( openshot::Timeline::GetFrame(long)        + 0xd9b )  [0x7fd6c028e43f]
  /usr/lib/x86_64-linux-gnu/libopenshot.so.16 ( openshot::PlayerPrivate::getFrame()       + 0x172 )  [0x7fd6c029d87e]
  /usr/lib/x86_64-linux-gnu/libopenshot.so.16 ( openshot::PlayerPrivate::run()            + 0x12e )  [0x7fd6c029cf1c]
  /usr/lib/x86_64-linux-gnu/libopenshot-audio.so.6 ( juce::Thread::threadEntryPoint()          + 0xa5  )  [0x7fd6bf3be705]
  /usr/lib/x86_64-linux-gnu/libopenshot-audio.so.6 ( threadEntryProc                           + 0x9   )  [0x7fd6bf3be8a9]
  /lib/x86_64-linux-gnu/libpthread.so.0 (                                           + 0x9609)  [0x7fd6cc744609]
  /lib/x86_64-linux-gnu/libc.so.6 ( clone                                     + 0x43  )  [0x7fd6cc880103]
---- End of Stack Trace ----
QObject::~QObject: Timers cannot be stopped from another thread

**Example 4**
preview_thread:INFO refreshFrame
preview_thread:INFO self.player.Position(): 649
Caught signal 11 (SIGSEGV)
---- Unhandled Exception: Stack Trace ----
free(): invalid pointer
Caught signal 6 (SIGABRT)
---- Unhandled Exception: Stack Trace ----
Segmentation fault (core dumped)

**System Details:**
v2.4.3 on Ubuntu 20.04.1

**Comment by ferdnyc** [comment]
@EQXTFL Version 2.4.3 is no longer supported, I'm afraid. Debian's packages are way out of date. The current release is OpenShot 2.5.1, and many of the effects have been extensively rewritten there so it's very unlikely the issue you're reporting still exists.

You can get an AppImage of either 2.5.1 or the Daily Build (current development code) from https://openshot.org/download/

**Comment by EQXTFL** [comment]
Confirmed fixed. @ferdnyc how can we ensure the Ubuntu repository is updated with the latest version?

**Comment by ferdnyc** [comment]
@EQXTFL 

Debian had bugs [925754](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=925754) and [925755](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=925755) tracking issues building current versions of the necessary libraries, but those issues were addressed and both bugs are now closed. If they still haven't updated, I have no idea what the holdup is. I'm not a Ubuntu (or any Debian-derived) distro user, so I have no real knowledge of their process.

Basically, you'd have to take it up with the package maintainers for your distro, using whatever process that would normally involve. If you open any new tickets with them to track the updates, please do share links to those requests in a new issue here, as well.

**Comment by EQXTFL** [comment]
I tried to reach out to the Ubuntu package maintainers but no request was answered. However, they updated to v2.5.1 on "Groovy Gorilla", which is current non-LongTermSupport version available. If anybody has trouble with openshot-qt on Ubuntu: please upgrade to this version or later.

https://launchpad.net/ubuntu/+source/openshot-qt
